### PR TITLE
Turn the linker options list in SM into a table

### PR DESCRIPTION
**Pull request by Veykril**
https://ferroussystems.clickup.com/t/8693tyf5g

### CODE REVIEW

**Review by tshepang — APPROVED** [review]
bors merge

File: ferrocene/doc/safety-manual/src/constraints.rst [hunk 1]
```diff
@@ -47,12 +47,6 @@ Constraints
      - The user shall not perform source modification when the build of the
        safety-related software is in progress.
      - :id:`AVD_PARALLEL_BUILD_006`
-   * - .. id:: CSTR_0070_TEST
```
Comment (pietroalbini) [inline ferrocene/doc/safety-manual/src/constraints.rst:50]: Why was this removed?

Comment (Veykril) [reply ferrocene/doc/safety-manual/src/constraints.rst:50]: The first bullet point talked about `Remove mention of ISO 26262`, I assumed it meant this?

Comment (pietroalbini) [reply ferrocene/doc/safety-manual/src/constraints.rst:50]: Oh, remove the mention in the same document ("A CLI option or one of its arguments may be outside the scope of the Ferrocene |iso_ref| qualification"), just having "..outside the scope of the Ferrocene qualification").
